librato/librato.py

```diff
@@ -19,16 +19,18 @@ def __init__(self, source, options):
         # self._url = build_url(source.get("user"), source.get("token"))
         self._meta_page = 0
         self._current_metric = 0
-        self._auth = base64.encodestring("%(user)s:%(token)s" % source).strip()
+        self._auth = base64.b64encode("%(user)s:%(token)s" % source).strip()
 
         # create the list of metrics along with their start timestamp
+        source_metrics = source.get("metrics", [])
```

Comment:
If `metrics` can be `None` (as you said in the description), then using `[]` as the default value for `.get` makes little sense due to the next line.
Its best to handle the "empty list as default" situation in a single place.

I think it makes more sense to have something like:
```python
source_metrics = source.get("metrics", None)
source_metrics = source_metrics if source_metrics is not None else []
```
(of course the `is not None` is optional, but personally I like how it reads, feel free to leave it out if you like)